mmdeploy/mmcv/ops/nms_match.py

```diff
@@ -0,0 +1,232 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+from torch import Tensor
+from torch.onnx import symbolic_helper as sym_help
+
+import mmdeploy
+from mmdeploy.core import mark
+
+
+class ONNXNMSMatchOp(torch.autograd.Function):
+    """Create onnx::NonMaxSuppressionMatch op.
+
+    NMS in mmcv only supports one class with no batch info. This class assists
+    in exporting NMS of ONNX's definition.
+    """
+
+    @staticmethod
+    def forward(ctx, boxes: Tensor, scores: Tensor, iou_threshold: float,
+                score_threshold: float) -> Tensor:
+        """Get NMS_Match_Fake output indices.
+
+        Args:
+            ctx (Context): The context with meta information.
+            boxes (Tensor): The bounding boxes of shape [N, num_boxes, 4].
+            scores (Tensor): The detection scores of shape
+                [N, num_boxes, num_classes].
+            max_output_boxes_per_class (int): Maximum number of output
+                boxes per class of nms.
+            iou_threshold (float): IOU threshold of nms.
+            score_threshold (float): score threshold of nms.
+
+        Returns:
+            Tensor: Selected indices of boxes. 2-D tensor of shape
+            (num_selected_indices, 3) with each row of
+            [batch_index, class_index, box_index].
+        """
+
+        from mmcv.ops import nms_match
+        batch_size, num_class, _ = scores.shape
+
+        indices = []
+        score_threshold = float(score_threshold)
+        iou_threshold = float(iou_threshold)
+        for batch_id in range(batch_size):
+            for cls_id in range(num_class):
+                _boxes = boxes[batch_id, ...]
+                # score_threshold=0 requires scores to be contiguous
+                _scores = scores[batch_id, cls_id, ...].contiguous()
+                # print(_boxes, _scores)
+                valid_mask = _scores > score_threshold
+                _boxes, _scores = _boxes[valid_mask], _scores[valid_mask]
```

Comment:
if we change `_boxes` inplace, then the `box_inds` returned by `nms_match` are not mapped to original `_boxes`